### ISSUE TITLE
[QUESTION] use dinov2 to train my own datasets

### ISSUE THREAD

**Issue by bighan123** [lightly-ai/lightly-train#392]
I want to use dinov3 to train my own datasets from scratch by your framework, could you please give the simple demo of how to organized my imgae data? or any docs I can consult?

**Comment by yutong-xiang-97** [comment]
Hi @bighan123, thank you for your interest in LightlyTrain!

May I know what kind of training task are you performing? We currently support distillation and semantic segmentation fine-tuning with DINOv3 backbones.

For how to organize your training data, you can refer to the "data" subsection in our docs for distillation and segmentation:

- Distillation: https://docs.lightly.ai/train/stable/train/index.html#data
- Semantic Segmentation: https://docs.lightly.ai/train/stable/semantic_segmentation.html#data

Hope this helps!

**Comment by bighan123** [comment]
thanks for your reply,  I don't want to finetune the dinov2 model on downstream tasks and I hope to train the 'new' dinov2 model by my datasets (millions of image data ), could lighty_train can help me to do that?

**Comment by yutong-xiang-97** [comment]
@bighan123 yes, LightlyTrain can definitely help you if you want to train DINOv3/v2 models with your own large-scale image data (no labels required!) You can use [the SSL pretraining / distillation methods we support](https://docs.lightly.ai/train/stable/methods/index.html#methods) mentioned here. The DINOv3 pretraining is not supported yet, but it is on our agenda :)

The data input format is exactly as specifed in the docs: https://docs.lightly.ai/train/stable/train/index.html#data. Simply providing you path to the image directory works out of the box for you :)

Hope this clarifies stuff!

**Comment by stegmuel** [comment]
Hi,

Yes lightly-train can help you pre-train a ViT using your own data and [DINOv2](https://docs.lightly.ai/train/stable/methods/dinov2.html): 
```
if __name__ == "__main__":
    lightly_train.train(
        out="out/my_experiment", 
        data="my_data_dir",
        model="dinov2/vitb14", # or "dinov2/vitb14-notpretrained" to start from scratch.
        method="dinov2",
    )
```
Images need to be placed in `my_data_dir`, but can be organized in different sub-folders within it.

**Comment by bighan123** [comment]
thanks for your reply and I have successfully trained the Dinov2 model and achieved satisfactory results, though several issues remain to be resolved. I wish to understand whether I can explicitly modify the dataset loading, processing, and training procedures to incorporate my own ideas. As your code is exceptionally well-structured, it has made it unclear where modifications should be made. Might you kindly provide relevant documentation for reference?

**Comment by guarin** [comment]
We have now example documentation on how to pretrain and fine-tune a DINOv2 model: https://docs.lightly.ai/train/stable/semantic_segmentation.html#pretrain-and-fine-tune-a-semantic-segmentation-model
